File: src/main/java/um/haberes/core/repository/InasistenciaDescuentoRepository.java
package um.haberes.core.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import um.haberes.core.kotlin.model.InasistenciaDescuento;

import java.util.Optional;

public interface InasistenciaDescuentoRepository extends JpaRepository<InasistenciaDescuento, Long> {

    Optional<InasistenciaDescuento> findByFacultadIdAndGeograficaIdAndDesdeGreaterThanEqualAndHastaLessThanEqual(
            Integer facultadId,
            Integer geograficaId,
            Integer inasistenciasMin,
            Integer inasistenciasMay
    );


}

File: src/main/java/um/haberes/core/kotlin/repository/InasistenciaDescuentoRepository.kt
package um.haberes.core.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import um.haberes.core.kotlin.model.InasistenciaDescuento
import java.util.Optional

interface InasistenciaDescuentoRepository : JpaRepository<InasistenciaDescuento?, Long?> {

    fun findByFacultadIdAndGeograficaIdAndDesdeGreaterThanEqualAndHastaLessThanEqual(
        facultadId: Int,
        geograficaId: Int,
        inasistencias: Int,
        inasistencias1: Int
    ): Optional<InasistenciaDescuento?>?;

}